sdk/appconfiguration/azure-spring-cloud-appconfiguration-config/src/main/java/com/azure/spring/cloud/config/implementation/ConnectionManager.java

```diff
@@ -0,0 +1,330 @@
+                    appProperties.getDefaultMinBackoff());
+                client.updateBackoffEndTime(Instant.now().plusNanos(backoffTime));
+                break;
+            }
+        }
+    }
+
+    /**
+     * Builds all the clients for a connection.
+     * @throws IllegalArgumentException when more than 1 connection method is given.
+     */
+    private void buildClients() {
+        // Single client or Multiple?
+        // If single call buildClient
+        int hasSingleConnectionString = StringUtils.hasText(configStore.getConnectionString()) ? 1 : 0;
+        int hasMultiEndpoints = configStore.getEndpoints().size() > 0 ? 1 : 0;
+        int hasMultiConnectionString = configStore.getConnectionStrings().size() > 0 ? 1 : 0;
+
+        if (hasSingleConnectionString + hasMultiEndpoints + hasMultiConnectionString > 1) {
+            throw new IllegalArgumentException(
+                "More than 1 Conncetion method was set for connecting to App Configuration.");
+        }
+
+        TokenCredential tokenCredential = null;
+
+        if (tokenCredentialProvider != null) {
+            tokenCredential = tokenCredentialProvider.getAppConfigCredential(configStore.getEndpoint());
+        }
+
+        boolean clientIdIsPresent = StringUtils.hasText(clientId);
+        boolean tokenCredentialIsPresent = tokenCredential != null;
+        boolean connectionStringIsPresent = configStore.getConnectionString() != null;
+
+        if ((tokenCredentialIsPresent || clientIdIsPresent)
+            && connectionStringIsPresent) {
+            throw new IllegalArgumentException(
+                "More than 1 Conncetion method was set for connecting to App Configuration.");
+        } else if (tokenCredential != null && clientIdIsPresent) {
+            throw new IllegalArgumentException(
+                "More than 1 Conncetion method was set for connecting to App Configuration.");
+        }
+
+        ConfigurationClientBuilder builder = getBuilder();
+        clients = new ArrayList<>();
+
+        if (configStore.getConnectionString() != null) {
+            client = buildClientConnectionString(configStore.getConnectionString(), builder);
+        } else if (configStore.getConnectionStrings().size() > 0) {
+            for (String connectionString : configStore.getConnectionStrings()) {
+                clients.add(buildClientConnectionString(connectionString, builder));
+            }
+        } else if (configStore.getEndpoints().size() > 0) {
+            for (String endpoint : configStore.getEndpoints()) {
+                clients.add(buildClientEndpoint(tokenCredential, endpoint, builder, clientIdIsPresent));
+            }
+        } else if (configStore.getEndpoint() != null) {
+            client = buildClientEndpoint(tokenCredential, configStore.getEndpoint(), builder, clientIdIsPresent);
+        }
+    }
+
+    private ConfigurationClientWrapper buildClientEndpoint(TokenCredential tokenCredential,
```

Comment:
The building of ConfigurationClient seems like not part of ConnectionManager's responsibility, we'd better move the build logic to a dedicated class.

Reply:
Done see `AppConfigurationReplicaClientBuilder`